tests/resiliency.rs

```diff
@@ -0,0 +1,183 @@
+        let rt = tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap();
+
+        let (proxies, toxic_local_addr) = create_proxies_and_configs();
+
+        let toxic_local_url = "tcp://".to_string() + &toxic_local_addr;
+        let toxic_proxy = &proxies[0];
+
+        let keys = vec!["key1", "key2", "key3"];
+        let values = vec!["value1", "value2", "value3"];
+        let kv: Vec<(&str, &str)> = keys.clone().into_iter().zip(values).collect();
+
+        let mut clean_client = rt.block_on(async {
+            async_memcached::Client::new("tcp://127.0.0.1:11211".to_string())
+                .await
+                .unwrap()
+        });
+
+        let mut toxic_client =
+            rt.block_on(async { async_memcached::Client::new(toxic_local_url).await.unwrap() });
+
+        let result = rt.block_on(async { clean_client.set_multi(kv.clone(), None, None).await });
+
+        assert!(result.is_ok());
+
+        let _ = toxic_proxy.with_down(|| {
+            rt.block_on(async {
+                let result = toxic_client.set_multi(kv.clone(), None, None).await;
+                assert_eq!(
+                    result,
+                    Err(async_memcached::Error::Io(
+                        std::io::ErrorKind::UnexpectedEof.into()
+                    ))
+                );
+            });
+        });
+    }
+
+    #[ignore = "Relies on a running memcached server and toxiproxy service"]
+    #[test]
+    fn test_set_multi_errors_with_toxiproxy_via_limit_data() {
+        // This test simulates a network error where the response from the server is cut short
+        let rt = tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap();
+
+        let (proxies, toxic_local_addr) = create_proxies_and_configs();
+
+        let toxic_local_url = "tcp://".to_string() + &toxic_local_addr;
+        let toxic_proxy = &proxies[0];
+        let keys = vec!["newkey1", "newkey2", "newkey3"];
+        let values = vec!["value1", "value2", "value3"];
+        let kv: Vec<(&str, &str)> = keys.clone().into_iter().zip(values.clone()).collect();
+
+        let mut clean_client = rt.block_on(async {
+            async_memcached::Client::new("tcp://127.0.0.1:11211".to_string())
+                .await
+                .unwrap()
+        });
+
+        let mut toxic_client =
+            rt.block_on(async { async_memcached::Client::new(toxic_local_url).await.unwrap() });
+
+        let result = rt.block_on(async { clean_client.set_multi(kv, None, None).await });
+
+        assert!(result.is_ok());
+
+        for key in &keys {
+            let _ = rt.block_on(async { clean_client.delete(key).await });
+            let result = rt.block_on(async { clean_client.get(key).await });
+            assert_eq!(result, Ok(None));
+        }
+
+        // Simulate a network error where a complete response is received for the first key but then
```

Comment:
This test kind of does it, but we don't really have a test where one key is successfully stored, then the next key fails to be stored, but the connection is still open? Like the first key sets because it is small, but the next key is large and hits a timeout.

Reply:
Ah, docs are helpful.  We were confused by the `stream` attribute in this test, but it controls directionality.  Using `"downstream"` as I did in this test is introducing toxicity from server => client.  If I run the same test with `"upstream"` then it will introduce the same toxicity from client => server.